torch/fx/graph.py

```diff
@@ -791,7 +792,7 @@ def add_global(name_hint: str, obj: Any):
 
             Returns: the global name that should be used to reference 'obj' in generated source.
             """
-            if _is_from_torch(obj):
+            if _is_from_torch(obj) and name_hint != 'device':  # to support registering torch.device
```

Comment:
This check is a little suspicious—`name_hint` does not have any defined meaning, it's just a hint for how to serialize and may be mangled, autogenerated to be numbers, whatever.

can you do a direct object check instead?

Reply:
ok :)